File: demo-apps/TradecoreDemoObjectiveC/TradecoreDemoObjectiveC/BaseVC/AdViewBaseVC/AdViewBaseViewController.m
//
//  AdViewBaseViewController.m
//  TradecoreDemoObjectiveC
//
//  Created by Olena Stepaniuk on 15.05.2025.
//

#import "AdViewBaseViewController.h"

@interface AdViewBaseViewController ()
@end

@implementation AdViewBaseViewController

- (NSString *)nibName {
    return @"AdViewBaseViewController";
}

- (instancetype)initWithAdSize:(CGSize)adSize {
    self = [super initWithNibName:self.nibName bundle:nil];
    if (self) {
        _adSize = adSize;
    }
    return self;
}

- (instancetype)init {
    return [self initWithAdSize:CGSizeMake(300, 250)];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.containerWidth.constant = self.adSize.width;
    self.containerHeight.constant = self.adSize.height;
}

- (void)onReloadTapped:(UIButton *)sender {
    [self.reloadButton setEnabled:false];
}

@end

File: demo-apps/TradecoreDemoAppSwift/TradecoreDemoAppSwift/BaseVC/AdViewBaseVC/AdViewBaseViewController.swift
//
//  BannerBaseViewController.swift
//  TradecoreDemoAppSwift
//
//  Created by Olena Stepaniuk on 14.05.2025.
//

import UIKit

fileprivate let adViewBaseViewControllerNibName = "AdViewBaseViewController"

class AdViewBaseViewController: UIViewController {
    
    @IBOutlet weak var containerView: UIView!
    
    @IBOutlet weak var containerWidth: NSLayoutConstraint!
    @IBOutlet weak var containerHeight: NSLayoutConstraint!
    
    @IBOutlet weak var reloadButton: UIButton!
    
    private(set) var adSize = CGSize(width: 300, height: 250)
    
    convenience init(adSize: CGSize) {
        self.init(nibName: adViewBaseViewControllerNibName, bundle: nil)
        self.adSize = adSize
    }
    
    convenience init() {
        self.init(nibName: adViewBaseViewControllerNibName, bundle: nil)
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        containerWidth.constant = adSize.width
        containerHeight.constant = adSize.height
    }
    
    @IBAction func onReloadTapped(_ sender: UIButton) {
        reloadButton.isEnabled = false
    }
}
